docs/config.md

```diff
@@ -22,37 +22,65 @@ You can use the `-C <config-file>` option to use a single configuration file and
+
+See {ref}`syntax-page` for more information about the Nextflow script syntax.
+
+### Assignments
+
+A config assignment consists of a config option and an expression separated by an equals sign:
 
 ```groovy
-name = value
+workDir = 'work'
+docker.enabled = true
+process.maxErrors = 10
 ```
 
-Please note, string values need to be wrapped in quotation characters while numbers and boolean values (`true`, `false`) do not. Also note that values are typed. This means that, for example, `1` is different from `'1'` — the former is interpreted as the number one, while the latter is interpreted as a string value.
+A config option consists of an *option name* prefixed by any number of *scopes* separated by dots. Config scopes are used to group related config options. See {ref}`config-options` for the full set of config options.
+
+The expression is typically a literal value such as a number, boolean, or string. However, any expression can be used:
 
-### Variables
+```groovy
+params.helper_file = "${projectDir}/assets/helper.txt"
+```
 
-Configuration properties can be used as variables in the configuration file by using the usual `$propertyName` or `${expression}` syntax.
+### Blocks
 
-For example:
+A config scope can also be specified as a block, which may contain multiple configuration options. For example:
 
 ```groovy
-propertyOne = 'world'
-anotherProp = "Hello $propertyOne"
-customPath = "$PATH:/my/app/folder"
+// dot syntax
+docker.enabled = true
+docker.runOptions = '-u $(id -u):$(id -g)'
+
+// block syntax
+docker {
+    enabled = true
+    runOptions = '-u $(id -u):$(id -g)'
+}
 ```
 
-Please note, the usual rules for {ref}`string-interpolation` are applied, thus a string containing a variable reference must be wrapped in double-quote chars instead of single-quote chars.
+As a result, deeply nested config options can be assigned in various ways. For example, the following three assignments are equivalent:
 
-The same mechanism allows you to access environment variables defined in the hosting system. Any variable name not defined in the Nextflow configuration file(s) is interpreted to be a reference to an environment variable with that name. So, in the above example, the property `customPath` is defined as the current system `PATH` to which the string `/my/app/folder` is appended.
+```groovy
+executor.retry.maxAttempt = 5
 
-### Comments
+executor {
+    retry.maxAttempt = 5
+}
 
-Configuration files use the same conventions for comments used by the Groovy or Java programming languages. Thus, use `//` to comment a single line, or `/*` .. `*/` to comment a block on multiple lines.
+executor {
+    retry {
+        maxAttempt = 5
+    }
+}
+```
 
 ### Includes
 
-A configuration file can include one or more configuration files using the keyword `includeConfig`. For example:
+A config file can include any number of other config files using the `includeConfig` keyword:
```

Comment:
Is not `configuration` better here?

Reply:
I guess either one is fine, but it's verbose to write configuration all the time, so we tend to write configuration the first time and config there after

Reply:
True that verbose, but this is a formal docs, "configuration" sounds better imo

Reply:
okay

Reply:
```suggestion
A configuration file can include any number of other configuration files using the `includeConfig` keyword:
```